orderbook/src/orderbook.rs

```diff
@@ -82,6 +84,9 @@ impl Orderbook {
             Some(order) => order,
             None => return Ok(AddOrderResult::InvalidSignature),
         };
+        if matches!(payload.from, Some(from) if from != order.order_meta_data.owner) {
```

Comment:
I guess I'm the only one who has trouble reading these `matches!` statements with `if` blocks inside. Guess, I will learn to cope with them 👨‍🎓